api/tests/test_tasks.py

```diff
@@ -54,30 +75,55 @@ def test_pdf_persistance(self, mock_client):
         order = serializer.validated_data
         create_delivery(order)
 
-        mock_delivery.print_pdf.assert_called_once()
+        mock_delivery.send_mail_to_vendor.assert_called_once()
 
     @patch('api.tasks.Client', autospec=True)
     def test_failure(self, mock_client):
         client = MagicMock()
-        client.create_delivery.return_value = Error({'mensaje': 'ERROR IN THE RECEIVED DATA'})
+        vendor = MagicMock()
+        delivery = Delivery({
+            "respuesta": "102",
+            "mensaje": "ERROR IN THE RECEIVED DATA", "numPedido": "test", "order_id": 33},
+            vendor)
+        client.create_delivery.return_value = [
+            delivery
+        ]
         mock_client.return_value = client
 
         serializer = OrderSerializer(data=self.data)
         self.assertTrue(serializer.is_valid())
         order = serializer.validated_data
-        result = create_delivery(order)
+        create_delivery(order)
+        self.assertTrue(delivery._is_errored())
 
-        self.assertEqual(result, {'error': 'ERROR IN THE RECEIVED DATA'})
-
-    @patch('api.tasks.Client', autospec=True)
-    def test_success(self, mock_client):
-        client = MagicMock()
-        client.create_delivery.return_value = MagicMock()
-        mock_client.return_value = client
+    @responses.activate
+    @patch('koiki.delivery.EmailMessage', autospec=True)
+    @patch('koiki.delivery.logger', autospec=True)
+    def test_create_delivery_sends_email(self, mock_logger, mock_email):
+        responses.add(responses.POST, 'https://testing_host/rekis/api/altaEnvios', status=200,
+                      json={
+                        'respuesta': '101',
+                        'mensaje': 'OK',
+                        'envios': [{'numPedido': '123', 'codBarras': 'yyy', 'etiqueta': 'abcd',
+                                    'respuesta': '101', 'mensaje': 'OK'}]
+                      })
 
         serializer = OrderSerializer(data=self.data)
         self.assertTrue(serializer.is_valid())
         order = serializer.validated_data
-        result = create_delivery(order)
 
-        self.assertEqual(result, {'success': 'all good'})
+        mock_email.send.return_value = True
+        context = {
+            'wcfmmp_host': koiki.wcfmmp_host,
+            'wc_order_id': 33,
+        }
+
+        message = render_to_string('contact_template.txt', context)
+        create_delivery(order)
+        mock_logger.info.assert_called_once_with(
+            "Sending Koiki pdf to email test@test.es")
+        mock_email.assert_called_once_with(
+            _("Enviament Koiki per a la comanda: {}").format(33),
+            message,
+            to=["test@test.es"]
+        )
```

Comment:
I think we go too far beyond the boundaries of what we should test at the integration level. The more we couple the test to the actual contents of the email, the hard it'll be to change things in the future. I would ensure the email is sent to the appropriate recipient. That's all I personally need to be confident about the code. 

What I miss are in-depth unit tests that could indeed cover the contents of the email. That's where this fits IMO.

Reply:
Tested in the new commits.